src/task.c

```diff
@@ -819,6 +816,21 @@ task_config_valid(const task_system_config *cfg)
    return STATUS_OK;
 }
 
+platform_status
+task_system_config_init(task_system_config *task_cfg,
+                        bool                use_stats,
+                        const uint64        num_bg_threads[NUM_TASK_TYPES],
+                        uint64              scratch_size)
+{
+   task_cfg->use_stats    = use_stats;
+   task_cfg->scratch_size = scratch_size;
+
+   memcpy(task_cfg->num_background_threads,
+          num_bg_threads,
+          NUM_TASK_TYPES * sizeof(num_bg_threads[0]));
+   return task_config_valid(task_cfg);
```

Comment:
Wouldn't it be better to see if `task_config_valid()` is true, and only then do the `memcpy()`? In other words, why not return immediately up front on L827 if `task_config_valid()` returns something other than `STATUS_OK`?

That way, we won't be polluting the output `task_cfg`'s contents in case of invalid configurations.